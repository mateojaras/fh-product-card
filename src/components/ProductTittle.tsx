import { useContext } from 'react';
import { ProductContext } from './ProductCard';
import styles from '../styles/styles.module.css';
import React from 'react';

export interface Props {
  tittle?: string;
  className?: string;
  style?: React.CSSProperties;
}

export const ProductTittle = ({ tittle, className, style }: Props) => {
  const { product } = useContext(ProductContext);
  return (
    <span className={`${styles.productDescription} ${className}`} style={style}>
      {tittle ? tittle : product.tittle}
    </span>
  );
};
