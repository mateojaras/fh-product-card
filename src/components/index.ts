import { ProductCardHOCProps } from '../interfaces/interfaces';
import { ProductButtons } from './ProductButtons';
import { ProductCard as ProductCardHOC } from './ProductCard';
import { ProductImage } from './ProductImage';
import { ProductTittle } from './ProductTittle';

export { ProductButtons } from './ProductButtons';
export { ProductImage } from './ProductImage';
export { ProductTittle } from './ProductTittle';

export const ProductCard: ProductCardHOCProps = Object.assign(ProductCardHOC, {
  Tittle: ProductTittle,
  Image: ProductImage,
  Buttons: ProductButtons,
});

export default ProductCard;
