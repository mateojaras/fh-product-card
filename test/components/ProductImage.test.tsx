import React from 'react';
import renderer from 'react-test-renderer';
import { ProductImage, ProductCard } from '../../src/components';
import { product2 } from '../data/products';

describe('ProductTitle', () => {
  test('debe de mostrar el componente correctamente con el titulo personalizado', () => {
    const wrapper = renderer.create(
      <ProductImage img="https://hola.jpg" className="custom-class" />
    );

    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  test('debe de mostrar el componente correctamente con el nombre del producto', () => {
    const wrapper = renderer.create(
      <ProductCard product={product2}>{() => <ProductImage />}</ProductCard>
    );

    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
