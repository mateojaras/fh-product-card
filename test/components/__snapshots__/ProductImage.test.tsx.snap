// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductTitle debe de mostrar el componente correctamente con el nombre del producto 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Product"
    className="productImg undefined"
    src="./coffee-mug.png"
  />
</div>
`;

exports[`ProductTitle debe de mostrar el componente correctamente con el titulo personalizado 1`] = `
<img
  alt="Product"
  className="productImg custom-class"
  src="https://hola.jpg"
/>
`;
